In [233]:
#Import packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sn


In [234]:
#Create DataFrame
df = pd.read_csv("Student Mental health.csv")

<h3>Data Exploration<h3>

In [235]:
#View first 5 rows of data
df.head()

Timestamp Choose your gender   Age What is your course?  \
0  8/7/2020 12:02             Female  18.0          Engineering   
1  8/7/2020 12:04               Male  21.0    Islamic education   
2  8/7/2020 12:05               Male  19.0                  BIT   
3  8/7/2020 12:06             Female  22.0                 Laws   
4  8/7/2020 12:13               Male  23.0         Mathemathics   

  Your current year of Study What is your CGPA? Marital status  \
0                     year 1        3.00 - 3.49             No   
1                     year 2        3.00 - 3.49             No   
2                     Year 1        3.00 - 3.49             No   
3                     year 3        3.00 - 3.49            Yes   
4                     year 4        3.00 - 3.49             No   

  Do you have Depression? Do you have Anxiety? Do you have Panic attack?  \
0                     Yes                   No                       Yes   
1                      No                  Yes                        No   
2                     Yes                  Yes                       Yes   
3                     Yes                   No                        No   
4                      No                   No                        No   

  Did you seek any specialist for a treatment?  
0                                           No  
1                                           No  
2                                           No  
3                                           No  
4                                           No

In [236]:
#View last 5 rows of data
df.tail()

Timestamp Choose your gender   Age What is your course?  \
96   13/07/2020 19:56:49             Female  21.0                  BCS   
97   13/07/2020 21:21:42               Male  18.0          Engineering   
98   13/07/2020 21:22:56             Female  19.0             Nursing    
99   13/07/2020 21:23:57             Female  23.0     Pendidikan Islam   
100  18/07/2020 20:16:21               Male  20.0   Biomedical science   

    Your current year of Study What is your CGPA? Marital status  \
96                      year 1        3.50 - 4.00             No   
97                      Year 2        3.00 - 3.49             No   
98                      Year 3        3.50 - 4.00            Yes   
99                      year 4        3.50 - 4.00             No   
100                     Year 2        3.00 - 3.49             No   

    Do you have Depression? Do you have Anxiety? Do you have Panic attack?  \
96                       No                  Yes                        No   
97                      Yes                  Yes                        No   
98                      Yes                   No                       Yes   
99                       No                   No                        No   
100                      No                   No                        No   

    Did you seek any specialist for a treatment?  
96                                            No  
97                                            No  
98                                            No  
99                                            No  
100                                           No

In [237]:
#
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101 entries, 0 to 100
Data columns (total 11 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   Timestamp                                     101 non-null    object 
 1   Choose your gender                            101 non-null    object 
 2   Age                                           100 non-null    float64
 3   What is your course?                          101 non-null    object 
 4   Your current year of Study                    101 non-null    object 
 5   What is your CGPA?                            101 non-null    object 
 6   Marital status                                101 non-null    object 
 7   Do you have Depression?                       101 non-null    object 
 8   Do you have Anxiety?                          101 non-null    object 
 9   Do you have Panic attack?                     101 non-null    obj

In [238]:
#Check null values if they exist 
df.isnull().sum()

Timestamp                                       0
Choose your gender                              0
Age                                             1
What is your course?                            0
Your current year of Study                      0
What is your CGPA?                              0
Marital status                                  0
Do you have Depression?                         0
Do you have Anxiety?                            0
Do you have Panic attack?                       0
Did you seek any specialist for a treatment?    0
dtype: int64

<h4>Display unique elements of columns<h4>

In [239]:
df['What is your course?'].unique()

array(['Engineering', 'Islamic education', 'BIT', 'Laws', 'Mathemathics',
       'Pendidikan islam', 'BCS', 'Human Resources', 'Irkhs',
       'Psychology', 'KENMS', 'Accounting ', 'ENM', 'Marine science',
       'KOE', 'Banking Studies', 'Business Administration', 'Law',
       'KIRKHS', 'Usuluddin ', 'TAASL', 'Engine', 'ALA',
       'Biomedical science', 'koe', 'Kirkhs', 'BENL', 'Benl', 'IT', 'CTS',
       'engin', 'Econs', 'MHSC', 'Malcom', 'Kop', 'Human Sciences ',
       'Biotechnology', 'Communication ', 'Diploma Nursing',
       'Pendidikan Islam ', 'Radiography', 'psychology', 'Fiqh fatwa ',
       'DIPLOMA TESL', 'Koe', 'Fiqh', 'Islamic Education', 'Nursing ',
       'Pendidikan Islam'], dtype=object)

In the list of courses provided there is Koe and koe which are the same courses but differently cased. So Lets fix that below and make the different word cases uniform.

In [240]:
#Change word casings from Capitalization of first letter to all upper case
df['What is your course?'] = df['What is your course?'].str.upper()
df['What is your course?'].unique()

array(['ENGINEERING', 'ISLAMIC EDUCATION', 'BIT', 'LAWS', 'MATHEMATHICS',
       'PENDIDIKAN ISLAM', 'BCS', 'HUMAN RESOURCES', 'IRKHS',
       'PSYCHOLOGY', 'KENMS', 'ACCOUNTING ', 'ENM', 'MARINE SCIENCE',
       'KOE', 'BANKING STUDIES', 'BUSINESS ADMINISTRATION', 'LAW',
       'KIRKHS', 'USULUDDIN ', 'TAASL', 'ENGINE', 'ALA',
       'BIOMEDICAL SCIENCE', 'BENL', 'IT', 'CTS', 'ENGIN', 'ECONS',
       'MHSC', 'MALCOM', 'KOP', 'HUMAN SCIENCES ', 'BIOTECHNOLOGY',
       'COMMUNICATION ', 'DIPLOMA NURSING', 'PENDIDIKAN ISLAM ',
       'RADIOGRAPHY', 'FIQH FATWA ', 'DIPLOMA TESL', 'FIQH', 'NURSING '],
      dtype=object)

In [241]:
df['Your current year of Study'].unique()

array(['year 1', 'year 2', 'Year 1', 'year 3', 'year 4', 'Year 2',
       'Year 3'], dtype=object)

From the list provided, year 1 and year 2 is the same as Year 1 and Year 2. There is a difference in word casing with one lower casing the first letter and the other upper casing the first letter. You need to make it uniform for correct data analysis. Will change that below

In [242]:
#Changing word casing from mixed uper and lower case, to lower case
df['Your current year of Study'] = df['Your current year of Study'].str.lower()
df['Your current year of Study'].unique()

array(['year 1', 'year 2', 'year 3', 'year 4'], dtype=object)

In [243]:
df['What is your CGPA?'].unique()

array(['3.00 - 3.49', '3.50 - 4.00', '3.50 - 4.00 ', '2.50 - 2.99',
       '2.00 - 2.49', '0 - 1.99'], dtype=object)

Note that the third value in the array, '3.50 - 4.00 ', contains an extra space at the end compared to the second value, '3.50 - 4.00'. This suggests that there might be inconsistent formatting in the data, with some values having extra spaces or other characters that are not consistent with the majority of values in the column. To fix inconsistent formatting in the data, you can use the string methods available in Pandas to clean and standardize the data in the column.

In [244]:
#the strip() method removes leading and trailing white spaces in each value of the column
df['What is your CGPA?'] = df['What is your CGPA?'].str.strip()

#the replace() method to replace any non-standard characters with the appropriate values.
df['What is your CGPA?'] = df['What is your CGPA?'].str.replace(' - ', '-')

#verify that the data in the column is now consistent and standardized.
df['What is your CGPA?'].unique()

array(['3.00-3.49', '3.50-4.00', '2.50-2.99', '2.00-2.49', '0-1.99'],
      dtype=object)

<h4>Replace Null Values<h4>

In [245]:
df['Age'].isnull().sum()

1

There is 1 missing age value. Calculate the Mean, Median and Mode of Column Age to figure out what best method you can use to fill the missing age value

In [246]:
#Calculate mean age 
mean_age = df['Age'].mean()
print(mean_age)

20.53


In [247]:
#Calculate median age
median_age = df['Age'].median()
print(median_age)

19.0


In [248]:
#Calculate mode age
mode_age = df['Age'].mode()
print(mode_age)

0    18.0
Name: Age, dtype: float64


Mean or Median Age: To replace null age values with the mean or median age of the non-null age values in the dataset. This method assumes that the age distribution is roughly normal and that the null values are missing at random.

Mode Age: to replace null age values with the mode age (most frequently occurring age) of the non-null age values. This method can work well if the age distribution is not normal and has a clear peak.

In [252]:
# Replace null age values with mode age (most frequently occurring age)
mode_age = df['Age'].mode()[0] # use [0] to get the first mode in case of multiple modes
df['Age'].fillna(mode_age, inplace=True)

#Check if there are still any missing values in the age column after filling
print(df['Age'].isnull().sum())


0


<h3>Exploratory Data Analysis<h3>

In [249]:
#Display Column Headers in Data Frame]
df.columns.values.tolist()

['Timestamp',
 'Choose your gender',
 'Age',
 'What is your course?',
 'Your current year of Study',
 'What is your CGPA?',
 'Marital status',
 'Do you have Depression?',
 'Do you have Anxiety?',
 'Do you have Panic attack?',
 'Did you seek any specialist for a treatment?']